pkg/provider/filters.go

```diff
@@ -115,3 +123,78 @@ func (env *TestEnvironment) GetHugepagesPods() []*Pod {
 	}
 	return filteredPods
 }
+
+func (env *TestEnvironment) GetCPUPinningPodsWithDpdk() []*Pod {
+	return filterCPUPinningPods(filterDPDKRunningPods(env))
+}
+
+func filterDPDKRunningPods(env *TestEnvironment) []*Pod {
+	var filteredPods []*Pod
+	const (
+		dpdkDriver           = "vfio-pci"
+		findDeviceSubCommand = "find . -name"
+	)
+	o := clientsholder.GetClientsHolder()
+	for _, pod := range env.Pods {
+		if len(pod.MultusPCIs) == 0 {
+			continue
+		} else {
+			ctx := clientsholder.NewContext(pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+			findCommand := fmt.Sprintf("%s '%s'", findDeviceSubCommand, pod.MultusPCIs[0])
+			outStr, errStr, err := o.ExecCommandContainer(ctx, findCommand)
+			if err != nil || errStr != "" {
+				logrus.Errorf("Failed to execute command %s in debug %s, errStr: %s, err: %s", findCommand, pod.String(), errStr, err)
+				continue
+			}
+			if strings.Contains(outStr, dpdkDriver) {
+				filteredPods = append(filteredPods, pod)
+			}
+		}
+	}
+	return filteredPods
+}
+
+func filterCPUPinningPods(pods []*Pod) []*Pod {
+	const (
+		verifyCPUAffinityCommand = "for pid in $(ps -A -o pid=); do taskset -pc $pid 2>/dev/null; done"
```

Comment:
Here you rely on the "taskset" program to be available in the CNF's container image. It might be installed on UBI-based images, but what if partners are using they're own images with file systems that do not have the taskset program installed? We can leave this implementation, but, at least, I recommend investigating if it's possible to use `nsenter` for this purpose.

Reply:
taskset is available in UBI image. Could you share me some partner image where I can verify again?

Reply:
After discussion between @greyerof @bnshr and myself, we agree to the following assumption for now: all guaranteed pods are assumed to be using cpu pinning. So to get the list of pods using cpu pinning is equivalent to the list of guaranteed pods ( already in the testenvironment struct).

Reply:
Oh great!  We get to re-use that slice.

Reply:
@bnshr , I don't understand your request. Partners may use whatever base image they want, but it's not possible to tell/predict which one in advance. They could also delete a lot of (not-used) binaries/programs from the base image just for the sake of reducing the size of the final container image.

Reply:
@edcdavid Is it true that these dpdk pods would be running with QoS as Guaranteed always? Or we need to verify that also from our end?

Reply:
I am not sure if dpdk pods need guaranteed qos to work at all, they should not if there are enough cores in the cpuset (as application is expecting) . However so far all DPDK apps in RAN space that I have seen needed the guaranteed QOS so that they can achieve their performance target. In my understanding the testcase itself is only applying to these guaranteed pods with extra modifications (remove interupts, etc) so as to really use 100% of the core with no noise. checking that we have DPDK in a guaranteed pod should be enough to identify pods to run this test in.

Reply:
Thanks!